Mathlib/Algebra/DirectLimit.lean

```diff
@@ -356,13 +411,70 @@ theorem lift_unique [IsDirected ι (· ≤ ·)] (F : DirectLimit G f →+ P) (x)
   · exact DirectLimit.induction_on x fun i x => by simp
 #align add_comm_group.direct_limit.lift_unique AddCommGroup.DirectLimit.lift_unique
 
+lemma lift_injective [IsDirected ι (· ≤ ·)]
+    (injective : ∀ i, Function.Injective <| g i) :
+    Function.Injective (lift G f P g Hg) := by
+  cases isEmpty_or_nonempty ι
+  · apply Function.injective_of_subsingleton
+  simp_rw [injective_iff_map_eq_zero] at injective ⊢
+  intros z hz
+  induction' z using DirectLimit.induction_on with _ g
+  rw [lift_of] at hz
+  rw [injective _ g hz, _root_.map_zero]
+
+variable {G'} (f) in
+/--
+For two family of abelian groups `G` and `G'` indexed by the same set with direct system `f` and
+`f'` respectively, if there is a compatible family of group-equivalence `eᵢ : Gᵢ ≅ Gᵢ'`, i.e. for
+`i ≤ j`, we have `eᵢ ∘ fᵢⱼ = fᵢⱼ' ∘ eⱼ`, there is a group equivalence `lim G ≅ lim G'`.
+-/
+def congr [IsDirected ι (· ≤ ·)] (equiv : (i : ι) → G i ≃+ G' i)
+    (compatible_equiv : ∀ i j h,
+      (equiv j).toAddMonoidHom.comp (f i j h) = (f' i j h).comp (equiv i).toAddMonoidHom) :
+    (DirectLimit G f) ≃+ DirectLimit G' f' where
```

Comment:
If you use `AddMonoidHom.toAddEquiv` here you can follow the same strategy you used for the linear map version.